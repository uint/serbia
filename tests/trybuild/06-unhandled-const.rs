use serbia::serbia;
use serde::{Deserialize, Serialize};

const BUFSIZE: usize = 300;

fn main() {
    // We do not handle constants unless they field is explicitly marked with
    // serde_bufsize.

    #[serbia]
    #[derive(Debug, Serialize, Deserialize, PartialEq)]
    struct S {
        arr_a: [u8; BUFSIZE],
        arr_b: [u8; 42],
        arr_small: [u8; 8],
    }

    let original = S {
        arr_a: [0; 300],
        arr_b: [0; 42],
        arr_small: [0; 8],
    };

    let serialized = serde_yaml::to_string(&original).unwrap();
    let deserialized = serde_yaml::from_str(&serialized).unwrap();

    assert_eq!(original, deserialized);
}