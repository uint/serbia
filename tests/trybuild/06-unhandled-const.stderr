error[E0277]: the trait bound `[u8; 300]: Serialize` is not satisfied
  --> $DIR/06-unhandled-const.rs:13:9
   |
13 |         arr_a: [u8; BUFSIZE],
   |         ^^^^^^^^^^^^^^^^^^^^ the trait `Serialize` is not implemented for `[u8; 300]`
   |
   = help: the following implementations were found:
             <[T; 0] as Serialize>
             <[T; 10] as Serialize>
             <[T; 11] as Serialize>
             <[T; 12] as Serialize>
           and 30 others
   = note: required by `main::_::_serde::ser::SerializeStruct::serialize_field`

error[E0277]: the trait bound `[u8; 300]: Deserialize<'_>` is not satisfied
  --> $DIR/06-unhandled-const.rs:13:9
   |
13 |         arr_a: [u8; BUFSIZE],
   |         ^^^^^^^^^^^^^^^^^^^^ the trait `Deserialize<'_>` is not implemented for `[u8; 300]`
   |
   = help: the following implementations were found:
             <&'a [u8] as Deserialize<'de>>
             <[T; 0] as Deserialize<'de>>
             <[T; 10] as Deserialize<'de>>
             <[T; 11] as Deserialize<'de>>
           and 30 others
   = note: required by `next_element`

error[E0277]: the trait bound `[u8; 300]: Deserialize<'_>` is not satisfied
  --> $DIR/06-unhandled-const.rs:13:9
   |
13 |         arr_a: [u8; BUFSIZE],
   |         ^^^^^^^^^^^^^^^^^^^^ the trait `Deserialize<'_>` is not implemented for `[u8; 300]`
   |
   = help: the following implementations were found:
             <&'a [u8] as Deserialize<'de>>
             <[T; 0] as Deserialize<'de>>
             <[T; 10] as Deserialize<'de>>
             <[T; 11] as Deserialize<'de>>
           and 30 others
   = note: required by `next_value`
